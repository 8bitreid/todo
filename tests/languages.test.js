const payloads = require('./fixtures/payloads')
const {gimmeRobot} = require('./helpers')

describe('languages', () => {
  const tree = [
    { path: 'go.go', sha: 'sha' },
    { path: 'ruby.rb', sha: 'sha' },
    { path: 'c.c', sha: 'sha' },
    { path: 'c-sharp.cs', sha: 'sha' }
  ]

  it('works with Go', async () => {
    const {robot, github} = gimmeRobot('noBlob.yml', {data: {items: [], total_count: 0}}, tree)
    const p = payloads.basic
    p.payload.commits[0].modified = ['go.go']

    await robot.receive(p)
    expect(github.issues.create).toHaveBeenCalledTimes(1)
    expect(github.issues.create.mock.calls[0]).toMatchSnapshot()
  })

  it('works with Ruby', async () => {
    const {robot, github} = gimmeRobot('noBlob.yml', {data: {items: [], total_count: 0}}, tree)

    const p = payloads.basic
    p.payload.commits[0].modified = ['ruby.rb']

    await robot.receive(p)
    expect(github.issues.create).toHaveBeenCalledTimes(1)
    expect(github.issues.create.mock.calls[0]).toMatchSnapshot()
  })

  it('works with C#', async () => {
    const {robot, github} = gimmeRobot('noBlob.yml', {data: {items: [], total_count: 0}}, tree)

    const p = payloads.basic
    p.payload.commits[0].modified = ['c-sharp.cs']

    await robot.receive(p)
    expect(github.issues.create).toHaveBeenCalledTimes(1)
    expect(github.issues.create.mock.calls[0]).toMatchSnapshot()
  })

  it('works with C', async () => {
    const {robot, github} = gimmeRobot('noBlob.yml', {data: {items: [], total_count: 0}}, tree)

    const p = payloads.basic
    p.payload.commits[0].modified = ['c.c']

    await robot.receive(p)
    expect(github.issues.create).toHaveBeenCalledTimes(1)
    expect(github.issues.create.mock.calls[0]).toMatchSnapshot()
  })
})
