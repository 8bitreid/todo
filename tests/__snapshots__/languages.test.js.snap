// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languages works with C 1`] = `
Array [
  Object {
    "assignee": "JasonEtco",
    "body": "###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Check that C works\\",\\"file\\":\\"c.c\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->",
    "labels": Array [
      "todo",
    ],
    "number": undefined,
    "owner": "JasonEtco",
    "repo": "test",
    "title": "Check that C works",
  },
]
`;

exports[`languages works with C# 1`] = `
Array [
  Object {
    "assignee": "JasonEtco",
    "body": "###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Check that C# works\\",\\"file\\":\\"c-sharp.cs\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->",
    "labels": Array [
      "todo",
    ],
    "number": undefined,
    "owner": "JasonEtco",
    "repo": "test",
    "title": "Check that C# works",
  },
]
`;

exports[`languages works with Go 1`] = `
Array [
  Object {
    "assignee": "JasonEtco",
    "body": "###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Check that Go works\\",\\"file\\":\\"go.go\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->",
    "labels": Array [
      "todo",
    ],
    "number": undefined,
    "owner": "JasonEtco",
    "repo": "test",
    "title": "Check that Go works",
  },
]
`;

exports[`languages works with Ruby 1`] = `
Array [
  Object {
    "assignee": "JasonEtco",
    "body": "###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Check that Ruby works\\",\\"file\\":\\"ruby.rb\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->",
    "labels": Array [
      "todo",
    ],
    "number": undefined,
    "owner": "JasonEtco",
    "repo": "test",
    "title": "Check that Ruby works",
  },
]
`;
